bundles/org.openhab.voice.actiontemplatehli/src/main/java/org/openhab/voice/actiontemplatehli/internal/ActionTemplateInterpreter.java

```diff
@@ -0,0 +1,1159 @@
+    }
+
+    public NLPInfo getNLPInfo(String text) throws IOException {
+        logger.debug("Processing: '{}'", text);
+        var nlpInfo = cacheMap.get(text);
+        if (nlpInfo == null) {
+            var tokens = tokenizeText(text);
+            var tags = languagePOSTagging(tokens);
+            var lemmas = languageLemmatize(tokens, tags);
+            nlpInfo = new NLPInfo(tokens, lemmas, tags);
+        } else {
+            logger.debug("resolved from cache");
+        }
+        logger.debug("tokens: {}", List.of(nlpInfo.tokens));
+        logger.debug("tags: {}", List.of(nlpInfo.tags));
+        logger.debug("lemmas: {}", List.of(nlpInfo.lemmas));
+        var finalNlpInfo = nlpInfo;
+        cacheMap.put(text, () -> finalNlpInfo);
+        return nlpInfo;
+    }
+
+    private @Nullable NLPInterpretationResult checkActionConfigs(String text, String[] tokens, String[] tags,
+            String[] lemmas) throws IOException {
+        // item defined actions have priority over type defined actions
+        var result = checkItemActions(text, tokens, tags, lemmas);
+        if (result != null) {
+            return result;
+        }
+        return checkTypeActionsConfigs(text, tokens, tags, lemmas);
+    }
+
+    private @Nullable NLPInterpretationResult checkItemActions(String text, String[] tokens, String[] tags,
+            String[] lemmas) throws IOException {
+        // Check item with action config
+        var itemsWithActions = getItemsWithActionConfigs();
+        Item targetItem = null;
+        ActionTemplateConfiguration targetActionConfig = null;
+        // store data to restore placeholder values
+        List<NLPPlaceholderData> placeholderValues = null;
+        // store span of dynamic placeholder, to invalidate others
+        Span dynamicSpan = null;
+        int matchScore = 0;
+        for (var entry : itemsWithActions.entrySet()) {
+            var actionConfigs = entry.getValue();
+            for (var actionConfig : actionConfigs) {
+                var templates = Arrays.stream(actionConfig.template.split(";")).map(String::trim)
+                        .collect(Collectors.toList());
+                for (var template : templates) {
+                    List<NLPPlaceholderData> currentPlaceholderValues = new ArrayList<>();
+                    var currentItem = entry.getKey();
+                    var scoreResult = getScoreWithPlaceholders(text, tokens, tags, lemmas, actionConfig, template,
+                            currentPlaceholderValues);
+                    if (scoreResult.score != 0 && scoreResult.score == matchScore) {
+                        if (targetItem == currentItem) {
+                            logger.warn("multiple alternative templates with same score '{}' item '{}'", template,
+                                    currentItem);
+                        } else {
+                            logger.warn("multiple templates with same score '{}' item '{}'", template, currentItem);
```

Comment:
These 2 logs are not at the right level. Probably DEBUG or INFO if this is very important for the user.

Reply:
I think it's important. It indicates a miss configuration because one of the actions will never be executed so one of them should be changed/removed by the user. I will change it to info.

Reply:
Looking at it again I still think it should be a warning as it indicates you have an unused action. 

But the message really looks like simple info right now, I will change it to indicate that you have something to fix. 

Is it ok if I keep it as a warning after this? or still is better to change it to info?

Reply:
We should just not flood the logs.
Keep a warning if you think it is really necessary for the user and this is not a regular log that will be logged at each call to the interpreter.